src/hotspot/share/gc/serial/serialBlockOffsetTable.hpp

```diff
@@ -215,190 +81,71 @@ class BlockOffsetSharedArray: public CHeapObj<mtGC> {
-  // cards are adjusted, but _unallocated_block isn't.
-  void single_block(HeapWord* blk_start, HeapWord* blk_end);
-  void single_block(HeapWord* blk, size_t size) {
-    single_block(blk, blk + size);
+  void set_offset_array(size_t index, HeapWord* high, HeapWord* low) {
+    assert(index < _vs.committed_size(), "index out of range");
+    assert(high >= low, "addresses out of order");
+    assert(pointer_delta(high, low) < BOTConstants::card_size_in_words(), "offset too large");
+    _offset_array[index] = checked_cast<u_char>(pointer_delta(high, low));
   }
 
-  // When the alloc_block() call returns, the block offset table should
-  // have enough information such that any subsequent block_start() call
-  // with an argument equal to an address that is within the range
-  // [blk_start, blk_end) would return the value blk_start, provided
-  // there have been no calls in between that reset this information
-  // (e.g. see BlockOffsetArrayNonContigSpace::single_block() call
-  // for an appropriate range covering the said interval).
-  // These methods expect to be called with [blk_start, blk_end)
-  // representing a block of memory in the heap.
-  virtual void alloc_block(HeapWord* blk_start, HeapWord* blk_end);
-  void alloc_block(HeapWord* blk, size_t size) {
-    alloc_block(blk, blk + size);
-  }
+  void set_offset_array(size_t left, size_t right, u_char offset) {
+    assert(right < _vs.committed_size(), "right address out of range");
+    assert(left <= right, "precondition");
+    size_t num_cards = right - left + 1;
 
-  // If true, initialize array slots with no allocated blocks to zero.
-  // Otherwise, make them point back to the front.
-  bool init_to_zero() { return _init_to_zero; }
-  // Corresponding setter
-  void set_init_to_zero(bool val) {
-    _init_to_zero = val;
-    assert(_array != nullptr, "_array should be non-null");
-    _array->set_init_to_zero(val);
+    fill_range(left, num_cards, offset);
   }
-
-  // Debugging
-  // Return the index of the last entry in the "active" region.
-  virtual size_t last_active_index() const = 0;
-  // Verify the block offset table
-  void verify() const;
-  void check_all_cards(size_t left_card, size_t right_card) const;
 };
 
-////////////////////////////////////////////////////////////////////////////
-// A subtype of BlockOffsetArray that takes advantage of the fact
-// that its underlying space is a ContiguousSpace, so that its "active"
-// region can be more efficiently tracked (than for a non-contiguous space).
-////////////////////////////////////////////////////////////////////////////
-class BlockOffsetArrayContigSpace: public BlockOffsetArray {
+// BlockOffsetTable divides the covered region into "N"-word subregions (where
+// "N" = 2^"LogN".  An array with an entry for each such subregion indicates
+// how far back one must go to find the start of the chunk that includes the
+// first word of the subregion.
+class BlockOffsetTable {
```

Comment:
The file has been renamed which is great, but the class is still called `BlockOffsetTable` without the prefix.